File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        Map<Character, Integer> map = new HashMap<>();
        char[] ss = s.toCharArray();
        char[] ts = t.toCharArray();
        for (int i = 0; i < ss.length; i++) {
            map.put(ss[i], map.getOrDefault(ss[i], 0) + 1);
        }
        for (int i = 0; i < ts.length; i++) {
            if (!map.containsKey(ts[i])) {
                return false;
            }
            map.put(ts[i], map.get(ts[i]) - 1);
            if (map.get(ts[i]) == 0) {
                map.remove(ts[i]);
            }
        }
        return map.size() == 0;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        char *cs = &s[0];
        char *ts = &t[0];
        unordered_map<char, int> map;
        for (size_t i = 0; i < s.length(); i++) {
            if (map.find(cs[i]) == map.end()) {
                map[cs[i]] = 1;
            } else {
                map[cs[i]] = map[cs[i]] + 1;
            }
        }
        for (size_t i = 0; i < t.length(); i++) {
            if (map.find(ts[i]) == map.end() || map[ts[i]] == 0) {
                return false;
            }
            map[ts[i]] = map[ts[i]] - 1;
            if (map[ts[i]] == 0) {
                map.erase(t[i]);
            }
        }
        return map.size() == 0;
    }
};